src/map/zone.cpp

```diff
@@ -290,37 +290,44 @@ void CZone::LoadZoneLines()
     }
 }
 
-/************************************************************************
-*                                                                       *
-*  Загружаем параметры погоды                                           *
-*                                                                       *
-************************************************************************/
+/*************************************************************************
+*                                                                        *
+*  Loads the weather for the zone from zone_bweather SQL Table           *
+*                                                                        *
+*  Weather is a rotating pattern of 2160 vanadiel days for each zone.    *
+*  It's stored as a blob of 2160 16-bit values, each representing 1 day  *
+*  starting from 0 and storing 3 5-bit weather values each.              *
+*                                                                        *
+*              0        00000       00000        00000                   *
+*              ^        ^^^^^       ^^^^^        ^^^^^                   *
+*          padding      normal      common       rare                    *
+*                                                                        *
+*************************************************************************/
 
 void CZone::LoadZoneWeather()
 {
     static const char* Query =
-        "SELECT "
-        "weather_day,"
-        "normal,"
-        "common,"
-        "rare "
-        "FROM zone_weather "
-        "WHERE zoneid = %u "
-        "ORDER BY weather_day";
+        "SELECT weather FROM zone_bweather WHERE zone = %u;";
 
     int32 ret = Sql_Query(SqlHandle, Query, m_zoneID);
-
     if (ret != SQL_ERROR && Sql_NumRows(SqlHandle) != 0)
     {
-        while (Sql_NextRow(SqlHandle) == SQL_SUCCESS)
-        {
-            m_WeatherVector.insert(std::make_pair((uint16)Sql_GetUIntData(SqlHandle, 0), zoneWeather_t(Sql_GetIntData(SqlHandle, 1),
-                Sql_GetIntData(SqlHandle, 2), Sql_GetIntData(SqlHandle, 3))));
-        }
+            Sql_NextRow(SqlHandle);
+            auto weatherBlob = reinterpret_cast<uint16*>(Sql_GetData(SqlHandle, 0));
+            for(uint16 i=1; i<WEATHER_CYCLE; i++)
```

Comment:
Would like some spaces in here:
`if (...`
`i = 1;`
etc.

Reply:
Updated to give you some space!